test.py

```diff
@@ -29,41 +27,45 @@ def error(msg):
 
     # Check for file existence
     if not os.path.isfile(jsonfile):
-        error("Needle '{}' is missing its JSON file!".format(needle))
+        error(f"Needle '{needle}' is missing its JSON file!")
         continue # parsing the json makes no sense
 
     if not os.path.isfile(pngfile):
-        error("Needle '{}' is missing its PNG file!".format(needle))
+        error(f"Needle '{needle}' is missing its PNG file!")
 
     # Check JSON content
     n = {}
     with open(jsonfile) as f:
         n = json.load(f)
 
     # Check if workaround tag exists if bugref is in name or if there is a reason in json file
-    for p in n.get('properties', []):
-        if isinstance(p, str) and p == 'workaround':
-            if not re.search(r'((poo|bsc|bnc|boo|kde)#?[A-Z0-9]+|jsc#?[A-Z]+-[0-9]+)', needle):
-                error("Needle '{}' includes a workaround tag but has no bug-ID in filename!".format(needle))
-            break
-        elif isinstance(p, dict) and p['name'] == 'workaround':
-            if p['value'] == '':
-                error("Needle '{}' includes a workaround tag but has no reason in json file!".format(needle))
-            break
+    properties = n.get('properties', [])
+    if properties is not None:
```

Comment:
This should never evaluate to false, as the `.get()` method returns a default of `[]` here.